### PR TITLE
feat: TWAP by Orbs

### PR DESCRIPTION
**Pull request by denis-orbs**
Integrate dTWAP by [Orbs](https://www.orbs.com/) into PancakeSwap.

[TWAP repo](https://github.com/orbs-network/twap)
[TWAP UI repo](https://github.com/orbs-network/twap-ui)

Screenshot 2024-03-04 at 15 33 32
Screenshot 2024-03-04 at 15 33 43

---

## PR-Codex overview
This PR introduces support for TWAP swaps and adds dynamic pages for TWAP and limit swaps. It also includes UI enhancements and new dependencies.

### Detailed summary
- Added `SwapType` enum for different swap types
- Updated routing for `/create/:currency*`
- Added TWAP-related functions and constants
- Updated package dependencies
- Implemented dynamic pages for TWAP and limit swaps
- Added `SwapSelection` component for swap type selection
- Enhanced UI components for TWAP and limit swaps

> The following files were skipped due to too many changes: `apps/web/src/views/Swap/Twap/Twap.tsx`, `pnpm-lock.yaml`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

**Comment by chef-jojo** [comment]
can we split those by page? You can use Component.Layout to preserve the layout

File: apps/web/src/views/Swap/TwapAndLimit.tsx [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+      </Wrapper>
+    </BodyWrapper>
+  )
+}
+
+export const TWAP_SUPPORTED_CHAINS = [56]
+
+export const isTwapSupported = (chainId?: number) => {
+  return TWAP_SUPPORTED_CHAINS.includes(chainId || 0)
+}
```
Comment (chefjackson) [inline apps/web/src/views/Swap/TwapAndLimit.tsx:229]: Can we use enum instead? It's from `@pancakeswap/chains`

```suggestion
export const TWAP_SUPPORTED_CHAINS = [ChainId.BSC]

export const isTwapSupported = (chainId?: ChainId) => {
  return TWAP_SUPPORTED_CHAINS.includes(chainId)
}
```

File: apps/web/src/views/Swap/index.tsx [hunk 2]
```diff
@@ -144,13 +165,35 @@ export default function Swap() {
...
             </StyledInputCurrencyWrapper>
           </StyledSwapContainer>
         </Flex>
       </Flex>
     </Page>
   )
 }
+
+const Menu = ({ swapType, onSelect }: { swapType: SwapType; onSelect: (value: number) => void }) => {
+  const { chainId } = useActiveWeb3React()
```
Comment (chefjackson) [inline apps/web/src/views/Swap/index.tsx:183]:
```suggestion
  const { chainId } = useActiveChainId()
```

File: apps/web/src/views/Swap/TwapAndLimit.tsx [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+
+  return {
+    isLoading: !value ? false : isLoading,
+    outAmount: value ? finalTrade?.outputAmount.numerator.toString() : '0',
+  }
+}
+
+const useUsd = (address?: string) => {
+  const currency = useCurrency(address)
+  const price = useStablecoinPrice(currency)
```
Comment (chefjackson) [inline apps/web/src/views/Swap/TwapAndLimit.tsx:79]:
```suggestion
  import { useCurrencyUsdPrice } from 'hooks/useCurrencyUsdPrice'
  const price = useCurrencyUsdPrice(currency)
```

File: apps/web/src/views/Swap/TwapAndLimit.tsx [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+  )
+
+  return onPresentCurrencyModal
+}
+
+export function TWAPPanel({ limit }: { limit?: boolean }) {
+  const { isDesktop } = useMatchBreakpoints()
+  const { account, chainId } = useWeb3React()
+  const tokens = useAllTokens()
+  const { connector } = useAccount()
```
Comment (chefjackson) [inline apps/web/src/views/Swap/TwapAndLimit.tsx:113]:
```suggestion
  const { chainId } = useActiveChainId()
  const tokens = useAllTokens()
  const { address: account, connector } = useAccount()
```

File: apps/web/src/views/Swap/TwapAndLimit.tsx [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+  )
+}
+
+export const OrderHistory = () => {
+  const { isDesktop } = useMatchBreakpoints()
+
+  return (
+    <BodyWrapper style={{ maxWidth: 'unset', marginTop: isDesktop ? 0 : 20 }}>
+      <Wrapper id="swap-page" style={{ padding: 0 }}>
+        <Orders />{' '}
```
Comment (chefjackson) [inline apps/web/src/views/Swap/TwapAndLimit.tsx:219]: Can we remove this space here?
```suggestion
        <Orders />
```

File: apps/web/src/views/Swap/V3Swap/containers/FormMain.tsx [hunk 6]
```diff
@@ -30,9 +30,17 @@ interface Props {
   tradeLoading?: boolean
   pricingAndSlippage?: ReactNode
   swapCommitButton?: ReactNode
+  loadedUrlParams: boolean
```
Comment (chefjackson) [inline apps/web/src/views/Swap/V3Swap/containers/FormMain.tsx:33]: Can we rename the variable for better understanding?

```suggestion
  isUrlParamsLoaded: boolean
```

**Comment by denis-orbs** [comment]
> can we split those by page? You can use Component.Layout to preserve the layout

you mean to do nested routing? 
twap page will be /swap/twap

so the modules will not load simultaneously?

**Comment by denis-orbs** [comment]
> can we split those by page? You can use Component.Layout to preserve the layout

image
Do you meant it to be like this (nested routes)? 

or did you want each tab to be in its own separate page, with a link in the navigation bar?
image

**Comment by chef-jojo** [comment]
> > can we split those by page? You can use Component.Layout to preserve the layout
> 
> image Do you meant it to be like this (nested routes)?
> or did you want each tab to be in its own separate page, with a link in the navigation bar? image

Yes, just we are just talking about having the url (page) for each tab. for the UI side, please follow @chef-bun comments

**Comment by chefjackson** [comment]
Hi the build failed. Could you help check? Seems like the lock file is not up-to-date

**Comment by chefjackson** [comment]
@SocketSecurity ignore npm/@sentry/node@5.30.0
@SocketSecurity ignore npm/@emotion/babel-plugin@11.11.0
@SocketSecurity ignore npm/@types/bignumber.js@5.0.0
@SocketSecurity ignore npm/@emotion/styled@11.11.0
@SocketSecurity ignore npm/@nomicfoundation/solidity-analyzer@0.1.1
@SocketSecurity ignore npm/@typechain/ethers-v5@10.2.1
@SocketSecurity ignore npm/@typechain/hardhat@6.1.6
@SocketSecurity ignore npm/@typechain/web3-v1@6.0.7
@SocketSecurity ignore npm/@nomiclabs/hardhat-etherscan@3.1.8
@SocketSecurity ignore npm/@emotion/serialize@1.1.3
@SocketSecurity ignore npm/@emotion/react@11.11.4
@SocketSecurity ignore npm/@defi.org/web3-candies@4.20.15
@SocketSecurity ignore npm/@nomicfoundation/edr@0.3.0

**Comment by chefjackson** [comment]
@SocketSecurity ignore npm/@defi.org/web3-candies@4.20.17
@SocketSecurity ignore npm/@nomicfoundation/edr@0.3.3
@SocketSecurity ignore npm/@emotion/styled@11.11.5
@SocketSecurity ignore npm/@emotion/serialize@1.1.4

**Comment by chefjackson** [comment]
@SocketSecurity ignore npm/@orbs-network/twap-ui@0.9.87

**Pull request merged**